File: go/docs/snippets/charge.client.go
//go:build ignore

// Client-side charge: pay a 402-gated endpoint in one retry.
//
// NewClient returns an *http.Client whose transport settles the 402 and
// replays the request with the payment credential. Mirrors the Client section
// of go/README.md. See ../../../docs/snippets-convention.md.
package main

import (
	"fmt"
	"io"

	"github.com/solana-foundation/pay-kit/go/paycore/solanatx"
	x402client "github.com/solana-foundation/pay-kit/go/protocols/x402/client"
)

func main() {
	var signer solanatx.Signer // your wallet signer
	var rpc solanatx.RPCClient // a Solana RPC client

	// snippet:start
	// The returned *http.Client settles a 402 transparently on any call.
	httpClient := x402client.NewClient(signer, rpc)

	resp, err := httpClient.Get("${URL}")
	if err != nil {
		panic(err)
	}
	defer resp.Body.Close()

	body, _ := io.ReadAll(resp.Body)
	fmt.Println(string(body))
	// snippet:end
}

File: python/docs/snippets/charge.client.py
# Client-side charge: pay an x402-gated endpoint in one retry.
#
# Mirrors examples/x402-client/main.py. See ../../../docs/snippets-convention.md.
import asyncio

from solana_pay_kit import Signer
from solana_pay_kit.protocols.x402.client import SolanaRpc, x402_async_client


async def main() -> None:
    signer = Signer.demo()  # or Signer.file("payer.json")
    rpc = SolanaRpc("https://api.devnet.solana.com")
    # snippet:start
    # x402_async_client returns an httpx.AsyncClient that turns any 402 into a
    # signed PAYMENT-SIGNATURE payment and replays the request — transparently.
    async with x402_async_client(signer, rpc) as http:
        resp = await http.get("${URL}")
        print(resp.status_code, resp.headers.get("payment-response"))
    # snippet:end


asyncio.run(main())
